src/sentry/incidents/grouptype.py

```diff
@@ -0,0 +1,32 @@
+from dataclasses import dataclass
+
+from sentry.incidents.utils.types import QuerySubscriptionUpdate
+from sentry.issues.grouptype import GroupCategory, GroupType
+from sentry.ratelimits.sliding_windows import Quota
+from sentry.types.group import PriorityLevel
+from sentry.workflow_engine.models import DataPacket
+from sentry.workflow_engine.models.detector import DetectorEvaluationResult, DetectorHandler
+
+
+# TODO: This will be a stateful detector when we build that abstraction
+class MetricAlertDetectorHandler(DetectorHandler[QuerySubscriptionUpdate]):
+    def evaluate(
+        self, data_packet: DataPacket[QuerySubscriptionUpdate]
+    ) -> list[DetectorEvaluationResult]:
+        # TODO: Implement
```

Comment:
how are you thinking this would handle the data condition groups to evaluate the detector conditions?

Reply:
I think we'd basically take the metric alerts logic and use it here, then use the data condition groups in place of the thresholds. I think that basically the logic here would be entirely custom.

We might end up abstracting it more to share with uptime, since they're similar systems.